l1-contracts/test/Rollup.t.sol

```diff
@@ -78,6 +79,178 @@ contract RollupTest is DecoderBase {
     _;
   }
 
+  function warpToL2Slot(uint256 _slot) public {
+    uint256 initialTime = rollup.GENESIS_TIME();
+    vm.warp(initialTime + _slot * Constants.AZTEC_SLOT_DURATION);
+  }
+
+  function testClaimEpochProofRight() public setUpFor("mixed_block_1") {
+    assertEq(rollup.getCurrentSlot(), 0, "genesis slot should be zero");
+
+    // make a quote that we'll update as we go to hit various errors
+    DataStructures.EpochProofQuote memory quote = DataStructures.EpochProofQuote({
+      signature: SignatureLib.Signature({isEmpty: false, v: 27, r: bytes32(0), s: bytes32(0)}),
+      epochToProve: 42,
+      validUntilSlot: 0,
+      bondAmount: 1,
+      rollup: address(0),
+      basisPointFee: 0
+    });
+
+    // sanity check that proven/pending tip are at genesis
+    vm.expectRevert(abi.encodeWithSelector(Errors.Rollup__NoEpochToProve.selector));
+    rollup.claimEpochProofRight(quote);
+
+    warpToL2Slot(1);
+    assertEq(rollup.getCurrentSlot(), 1, "warp to slot 1 failed");
+    assertEq(rollup.getCurrentEpoch(), 0, "Invalid current epoch");
+
+    // empty slots do not move pending chain
+    vm.expectRevert(abi.encodeWithSelector(Errors.Rollup__NoEpochToProve.selector));
+    rollup.claimEpochProofRight(quote);
+
+    _testBlock("mixed_block_1", false, 1);
+    assertEq(rollup.getCurrentEpoch(), 0, "Invalid current epoch");
+    assertEq(rollup.getCurrentSlot(), 1, "propose block should not shift slot");
+    assertEq(rollup.getPendingBlockNumber(), 1, "Invalid pending block number");
+    assertEq(rollup.getProvenBlockNumber(), 0, "Invalid proven block number");
+    assertEq(rollup.getEpochToProve(), 0, "Invalid epoch to prove");
+
+    // the quote has the wrong epoch
+    vm.expectRevert(
+      abi.encodeWithSelector(Errors.Rollup__NotClaimingCorrectEpoch.selector, 0, quote.epochToProve)
+    );
+    rollup.claimEpochProofRight(quote);
+    quote.epochToProve = 0;
```

Comment:
The way that the quote is altered makes it harder to follow. Would be easier if the `quote` was passing, and then we see test setting to something not passing and causing it to fail. Something that seems to fit nicely for separate tests.